src/overlays/actors/ovl_Bg_Haka_Bombwall/z_bg_haka_bombwall.c

```diff
@@ -28,50 +36,191 @@ const ActorInit Bg_Haka_Bombwall_InitVars = {
     (ActorFunc)BgHakaBombwall_Draw,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_80BD64A0 = {
-    { COLTYPE_NONE, AT_NONE, AC_ON | AC_TYPE_PLAYER, OC1_NONE, OC2_NONE, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0x00000008, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_NONE, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_NONE,
+        OC2_NONE,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000008, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_NONE,
+    },
     { 80, 80, 0, { 0, 0, 0 } },
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80BD64D4[] = {
+static s16 D_80BD64CC[4] = { 24, 15, 10, 5 };
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 500, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
-
-extern ColliderCylinderInit D_80BD64A0;
-extern InitChainEntry D_80BD64D4[];
-
-extern UNK_TYPE D_06000040;
-extern UNK_TYPE D_06000148;
-extern UNK_TYPE D_06001680;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Haka_Bombwall/func_80BD5E00.s")
-
+s32 func_80BD5E00(BgHakaBombwall* this) {
+    if (this->collider.base.acFlags & AC_HIT) {
+        if ((this->collider.base.ac != NULL) &&
+            (Math3D_Vec3fDistSq(&this->dyna.actor.world.pos, &this->collider.base.ac->world.pos) < SQ(80.0f))) {
+            return true;
+        }
+    }
+    return false;
+}
+
+#ifdef NON_MATCHING
+void func_80BD5E6C(BgHakaBombwall* this, PlayState* play) {
+    s32 i;
+    Vec3f spE0;
+    Vec3f spD4;
+    Vec3f spC8;
+    Vec3f spBC;
+    f32 temp_fs0;
+    f32 phi_fs1;
+    s16 phi_s0;
+    s16 gravity;
+    s16 phi_t0;
+
+    Matrix_Push();
+    Matrix_RotateYS(this->dyna.actor.shape.rot.y, MTXMODE_NEW);
+    temp_fs0 = 0.0f;
+    phi_fs1 = 0.0f;
+
+    for (i = 0; i < 30; i++) {
+        temp_fs0 += 60.0f + (Rand_ZeroOne() * 20.0f);
+        if (temp_fs0 > 75.0f) {
+            temp_fs0 -= 150.0f;
+        }
+        spC8.x = temp_fs0;
+
+        phi_fs1 += 5;
+        spC8.y = phi_fs1;
+
+        spC8.z = (Rand_ZeroOne() * 20.0f) - 10.0f;
+
+        spBC.x = ((Rand_ZeroOne() - 0.5f) * 5.0f) + (temp_fs0 * 0.053333335f);
```

Comment:
Maybe `4.0f / 75.0f`?
```suggestion
        spBC.x = ((Rand_ZeroOne() - 0.5f) * 5.0f) + (temp_fs0 * (4.0f / 75.0f));
```